src/tbb/task_dispatcher.cpp

```diff
@@ -22,6 +22,8 @@ namespace detail {
 namespace r1 {
 
 static inline void spawn_and_notify(d1::task& t, arena_slot* slot, arena* a) {
+    a->pool_mask[this_task_arena::current_thread_index()].store(
```

Comment:
Don't we want to set this flag after the `spawn`?

Reply:
Pool might be empty just after spawn.